shared/js/background/dynamic-rule-id.js

```diff
@@ -0,0 +1,40 @@
+const browserWrapper = require('./wrapper.es6')
+const DYNAMIC_RULE_ID_START = 100000
+const DYNAMIC_RULE_STORAGE_KEY = 'dynamicRuleOffset'
+let dynamicRuleOffset
+
+export function getDynamicRuleId () {
+    const nextRuleId = DYNAMIC_RULE_ID_START + dynamicRuleOffset
+    dynamicRuleOffset += 1
+    browserWrapper.setToSessionStorage(DYNAMIC_RULE_STORAGE_KEY, dynamicRuleOffset)
+    return nextRuleId
+}
+
+async function _getExistingRulesAssigned () {
+    let rules = []
+
+    // get all declarative rules
+    const set1 = await chrome.declarativeNetRequest.getDynamicRules()
+    rules = rules.concat(set1.filter(r => r.id >= DYNAMIC_RULE_ID_START))
+    const set2 = await chrome.declarativeNetRequest.getSessionRules()
+
+    // get all rules that were already dynamically created
+    rules = rules.concat(set2.filter(r => r.id >= DYNAMIC_RULE_ID_START))
+    return rules
+}
+
+// set offset and update based on rules that have been created
+(async () => {
```

Comment:
Might as well call this Promise `ready` and export it, so that other modules can wait for this to be ready if necessary.